iam.tf

```diff
@@ -0,0 +1,55 @@
+resource aws_iam_role main {
+  name = format("%s-Role", local.name)
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": [
+            "autoscaling.amazonaws.com",
+            "ec2.amazonaws.com"
+        ]
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+# TODO: This is awful, take the real bucket or make one and use that here
```

Comment:
Yes I agree. We can use a `count` here to toggle the creation depending on an input s3 bucket.